### PR TITLE
[RW-9675][risk=low] Checks request and block accessing default app engine hostname is not cloudtask or cron jobs

### PR DESCRIPTION
**Pull request by yonghaoy**
Description:
https://docs.google.com/document/d/1hhGvLLrwi_tQojUuQ7DDJvGDz525-u8rAiyiSkHHS3w/edit#heading=h.iflteer64u9l

Tested and verified cron job, cloud task and regular api working in test

---
**PR checklist**

- [ ] This PR meets the Acceptance Criteria in the JIRA story
- [ ] The JIRA story has been moved to Dev Review
- [ ] This PR includes appropriate unit tests
- [ ] I have added explanatory comments where the logic is not obvious
- [ ] I have run and tested this change locally, and my testing process is described here
- [ ] If this includes a new feature flag, I have created and linked new JIRA tickets to (a) turn on the feature flag and (b) remove it later
- [ ] If this includes an API change, I have run the E2E tests on this change against my local server with [yarn test-local](https://github.com/all-of-us/workbench/blob/master/e2e/README.md#examples) because this PR won't be covered by the CircleCI tests 
- [ ] If this includes a UI change, I have taken screen recordings or screenshots of the new behavior and notified the PO and UX designer in [Slack](https://pmi-engteam.slack.com/archives/C02MWP2RN5P)
- [ ] If this change impacts deployment safety (e.g. removing/altering APIs which are in use) I have documented these in the description
- [ ] If this includes an API change, I have updated the appropriate Swagger definitions and updated the appropriate API consumers
  * AoU UI
  * [Perf tests](https://github.com/broadinstitute/mcnulty/blob/develop/src/test/scala/services/AoU.scala)
  * [Researcher Directory export](https://github.com/all-of-us/workbench/wiki/Researcher-Directory-(RDR-export))
  * Cron tasks - for Offline*Controllers
  * SumoLogic - for EgressAlert

### CODE REVIEW

File: api/src/test/java/org/pmiops/workbench/interceptors/AuthInterceptorTest.java [hunk 1]
```diff
@@ -280,4 +286,29 @@ public void authorityCheckPermitsWhenUserHasAuthority() throws Exception {
...
+    when(mockRequest.getMethod()).thenReturn(HttpMethods.GET);
+
+    when(mockRequest.getRequestURL()).thenReturn(new StringBuffer("domain"));
+
+    assertThat(interceptor.preHandle(mockRequest, mockResponse, mockHandler)).isFalse();
+    verify(mockResponse).sendError(HttpServletResponse.SC_FORBIDDEN);
+  }
+
+  @Test
+  public void preHandle_apiBaseUrlNotMatch_cloudTask_allowed() throws Exception {
```
Comment (jmthibault79) [inline api/src/test/java/org/pmiops/workbench/interceptors/AuthInterceptorTest.java:302]: test for Cron as well?

Comment (yonghaoy) [reply api/src/test/java/org/pmiops/workbench/interceptors/AuthInterceptorTest.java:302]: We are not adding checks for Cron job. Cron job does not have user bearer token and aou_auth, it returns true before reaching to the new check: https://github.com/all-of-us/workbench/blob/7a37a332a0f6c8e833af46f217e89a880dad8086/api/src/main/java/org/pmiops/workbench/interceptors/AuthInterceptor.java#L105

Seems our existing `CronInterceptorTest` has good coverage already

File: api/src/main/java/org/pmiops/workbench/interceptors/AuthInterceptor.java [hunk 2]
```diff
@@ -107,6 +107,21 @@ public boolean preHandle(HttpServletRequest request, HttpServletResponse respons
 
     String authorizationHeader = request.getHeader(HttpHeaders.AUTHORIZATION);
 
+    if (!request
+            .getRequestURL()
+            .toString()
+            .startsWith(workbenchConfigProvider.get().server.apiBaseUrl)
+        && !InterceptorUtils.isCloudTaskRequest(apiOp)) {
+      // Is authentication is required and not cloud task, the request should send to API external
```
[the remark this answers is not in the document]
Comment (yonghaoy) [reply api/src/main/java/org/pmiops/workbench/interceptors/AuthInterceptor.java:115]: I was struggling how to explain this. Just updated it a bit, and probably need your help to polish this. (especially in Cloud task section)

File: api/src/main/java/org/pmiops/workbench/interceptors/AuthInterceptor.java [hunk 3]
```diff
@@ -107,6 +107,26 @@ public boolean preHandle(HttpServletRequest request, HttpServletResponse respons
 
     String authorizationHeader = request.getHeader(HttpHeaders.AUTHORIZATION);
 
+    if (!request
+            .getRequestURL()
+            .toString()
+            .startsWith(workbenchConfigProvider.get().server.apiBaseUrl)
+        && !InterceptorUtils.isCloudTaskRequest(apiOp)) {
+      // API backend server has two URL:
+      // 1: Default appshot hostname. Cron job and cloud task have to use this URL.
```
Comment (jmthibault79) [inline api/src/main/java/org/pmiops/workbench/interceptors/AuthInterceptor.java:116]: appSPOT

Comment (jmthibault79) [reply api/src/main/java/org/pmiops/workbench/interceptors/AuthInterceptor.java:116]: This is great otherwise.  Thanks for the update 👍

**Review by jmthibault79 — APPROVED** [review]

**Pull request merged**